spec/models/course/assessment_spec.rb

```diff
@@ -17,6 +17,42 @@
     let(:assessment) { create(:assessment, *assessment_traits, course: course) }
     let(:assessment_traits) { [] }
 
+    describe 'validations' do
+      context 'when it is not in draft' do
+        context 'when it has no question' do
+          subject { build(:assessment, draft: false) }
+          it { is_expected.to_not be_valid }
+        end
+
+        context 'when it has questions' do
+          subject do
+            assessment = build(:assessment)
+            create_list(:course_assessment_question_multiple_response, 3, assessment: assessment)
+            assessment.draft = true
+            assessment
+          end
+          it { is_expected.to be_valid }
+        end
+      end
+
+      context 'when it is in draft' do
+        context 'when it has no question' do
+          subject { build(:assessment, draft: true) }
+          it { is_expected.to be_valid }
+        end
+
+        context 'when it has questions' do
+          subject do
+            assessment = build(:assessment)
```

Comment:
you can use `:with_all_question_types` then you won't need to generate a separate list of questions